lib/panomosity/generalized_neighborhood.rb

```diff
@@ -101,9 +112,14 @@ def calculate_similar_neighborhoods(type: :horizontal, count: 3)
 
       def std_outlier_reduction(type: :horizontal, max_reduction_attempts: 2, reduction_attempts: 0)
         return if reduction_attempts >= max_reduction_attempts
+
         logger.debug "twice reducing #{type} neighborhood std outliers"
-        avg, std = *calculate_average_and_std(values: similar_neighborhoods(type: type).map(&:dist_std))
-        similar_neighborhoods(type: type).select! { |n| (avg - n.dist_std).abs <= std }
+
+        neighborhoods = similar_neighborhoods! type: type
```

Comment:
Also sorry to be that guy, but use parenthesis here

Reply:
We've been making an exception for method calls ending in `!` and `?` to improve readability.
They shouldn't ever get chained, either, so it's less of an issue.

https://github.com/elevatesystems/stitches_rails/blob/develop/.rubocop.yml#L117

I'll change it though.
Besides, that's not the convention in this repo: https://github.com/elevatesystems/panomosity/blob/ca3a2f552cd4112a89c032c863a7f5c25f1e311d/lib/panomosity/runner.rb#L61

Reply:
#ThatGuyGarcia